src/librustc/middle/lang_items.rs

```diff
@@ -158,6 +158,12 @@ impl<'a, 'v, 'tcx> Visitor<'v> for LanguageItemCollector<'a, 'tcx> {
 
             if let Some(item_index) = item_index {
                 self.collect_item(item_index, self.ast_map.local_def_id(item.id))
+            } else {
+                let item_def_id = self.ast_map.local_def_id(item.id);
+                let span = self.ast_map.span_if_local(item_def_id).unwrap();
```

Comment:
why not call `self.ast_map.span(item.id)`? Then you can skip creating the `item_def_id` too.